File: Practica_2/Servicio/src/servicio/comun/TDatosRepositorio.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package servicio.comun;

import java.io.Serializable;
import java.util.Objects;

/**
 *
 * @author Usuario
 */
public class TDatosRepositorio implements Serializable{
    
    private int numeroLibros;
    private String nombreRepositorio;
    private String direccionRepositorio;
    private RepositorioLibro repositorioLibro;

    public TDatosRepositorio() {
    }

    public TDatosRepositorio(int numeroLibros, String nombreRepositorio, String direccionRepositorio, RepositorioLibro repositorioLibro) {
        this.numeroLibros = numeroLibros;
        this.nombreRepositorio = nombreRepositorio;
        this.direccionRepositorio = direccionRepositorio;
        this.repositorioLibro = repositorioLibro;
    }

    public int getNumeroLibros() {
        return numeroLibros;
    }

    public void setNumeroLibros(int numeroLibros) {
        this.numeroLibros = numeroLibros;
    }

    public String getNombreRepositorio() {
        return nombreRepositorio;
    }

    public void setNombreRepositorio(String nombreRepositorio) {
        this.nombreRepositorio = nombreRepositorio;
    }

    public String getDireccionRepositorio() {
        return direccionRepositorio;
    }

    public void setDireccionRepositorio(String direccionRepositorio) {
        this.direccionRepositorio = direccionRepositorio;
    }

    public RepositorioLibro getRepositorioLibro() {
        return repositorioLibro;
    }

    public void setRepositorioLibro(RepositorioLibro repositorioLibro) {
        this.repositorioLibro = repositorioLibro;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 89 * hash + this.numeroLibros;
        hash = 89 * hash + Objects.hashCode(this.nombreRepositorio);
        hash = 89 * hash + Objects.hashCode(this.direccionRepositorio);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final TDatosRepositorio other = (TDatosRepositorio) obj;
        if (this.numeroLibros != other.numeroLibros) {
            return false;
        }
        if (!Objects.equals(this.nombreRepositorio, other.nombreRepositorio)) {
            return false;
        }
        return Objects.equals(this.direccionRepositorio, other.direccionRepositorio);
    }
    
    
}

File: Practica_3/ServicioBiblioteca/ServicioBiblioteca/TDatosRepositorio.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServicioBiblioteca
{
    [Serializable]
    public class TDatosRepositorio
    {
        private int numeroLibros;
        private String nombreRepositorio;
        private String direccionRepositorio;
        private String rutaRepositorio;
        private RepositorioLibroIntf repositorioLibro;

        public TDatosRepositorio()
        {
        }

        public TDatosRepositorio(int numeroLibros, String nombreRepositorio, String direccionRepositorio, RepositorioLibroIntf repositorioLibro, String rutaRepositorio)
        {
            this.numeroLibros = numeroLibros;
            this.nombreRepositorio = nombreRepositorio;
            this.direccionRepositorio = direccionRepositorio;
            this.repositorioLibro = repositorioLibro;
            this.rutaRepositorio = rutaRepositorio;
        }
        public string NombreRepositorio
        {
            get { return this.nombreRepositorio; }
            set { this.nombreRepositorio = value; }
        }

        public string DireccionRepositorio
        {
            get { return this.direccionRepositorio; }
            set { this.direccionRepositorio = value; }
        }

        public int NumeroLibros
        {
            get { return this.numeroLibros; }
            set { this.numeroLibros = value; }
        }

        public RepositorioLibroIntf RepositorioLibro
        {
            get { return this.repositorioLibro; }
            set { this.repositorioLibro = value; }
        }

        public String RutaRepositorio
        {
            get { return this.rutaRepositorio; }
            set { this.rutaRepositorio = value; }
        }


        public override int GetHashCode()
        {
            int hash = 7;
            hash = 89 * hash + this.numeroLibros.GetHashCode();
            hash = 89 * hash + (this.nombreRepositorio?.GetHashCode() ?? 0);
            hash = 89 * hash + (this.direccionRepositorio?.GetHashCode() ?? 0);
            return hash;
        }


        public override bool Equals(Object obj)
        {
            if (this == obj)
            {
                return true;
            }
            if (obj == null)
            {
                return false;
            }
            if (GetType() != obj.GetType())
            {
                return false;
            }

            TDatosRepositorio other = (TDatosRepositorio)obj;

            if (this.numeroLibros != other.numeroLibros)
            {
                return false;
            }
            if (!Equals(this.nombreRepositorio, other.nombreRepositorio))
            {
                return false;
            }
            return Equals(this.direccionRepositorio, other.direccionRepositorio);
        }
    }
}
